networkx/algorithms/d_separation.py

```diff
@@ -312,12 +316,36 @@ def is_minimal_d_separator(G, u, v, z):
     v : node
         A node in the graph.
     z : Set of nodes
-        The set of nodes to check if it is a minimal d-separating set.
+        The set of nodes to check if it is a minimal d-separating set. If
+        `check_d_sep = True`, then a call to :func:`d_separated` is called
+        to verify that `z` is in fact a d-separator. If `check_d_sep = False`,
+        then no check will be done, and it will be assumed that the user
+        has already verified that `z` is a d-separator.
+    check_d_sep : bool
+        Whether or not to first check if `z` is a d-separating set. By default,
+        it is `True`. Set to `False` to turn off the d-separating set check.
 
     Returns
     -------
     bool
-        Whether or not the `z` separating set is minimal.
+        Whether or not the set `z` is a d-separator and is also minimal.
+
+    Examples
+    --------
+    >>> G = nx.path_graph([0, 1, 2, 3], create_using=nx.DiGraph)
+    >>> G.add_node(4)
+    >>> # since {1} is the minimal d-separator, this will return False
+    >>> nx.is_minimal_d_separator(G, 0, 2, {1,3,4})
+    False
+    >>>
+    >>> # if we already know that z is a d-separator, we can turn off the check
+    >>> nx.is_minimal_d_separator(G, 0, 2, {1}, False)
+    True
+    >>> # if we do not know that z is a d-separator, then we can first check it ourselves
+    >>> is_d_sep = nx.d_separated(G, {0}, {4}, {1})
+    >>> is_minimal = nx.is_minimal_d_separator(G, 0, 2, {1}, False)
+    >>> print(is_d_sep and is_minimal)
+    True
```

Comment:
I think the code for checking both d_separator and minimal is simpler than what appears here. That's why I'm thinking it might be better to avoid the keyword argument. It would be different if the algorithm for minimal and for d_separated were more similar. But it feels uncomfortable to try to join two functions that are really asking different questions.
```python
# this seems straightforward to me
if nx.d_separated(G, {u}, {v}, z) and nx.is_minimal_d_separator(G, u, v, z):
    print("This d_separator is minimal")
```
Would a better name for `is_minimal_d_separator` solve the problem?

Or... Is there some way to use the algorithm for "minimal" and as part of that algorithm check that z is a d_separator?

Reply:
> Or... Is there some way to use the algorithm for "minimal" and as part of that algorithm check that z is a d_separator?

Unfortunately, not from what I see. IIUC, the implicit assumption for `is_minimal_d_separator` is that a set is already a d-separator. But Algorithm 1 seems to have the check of d-separator internally: https://ftp.cs.ucla.edu/pub/stat_ser/r254.pdf. E.g. in Theorem 5 d-separation is assumed apriori. But Theorem 4 has a check for d-separation explicitly.

We could implement algo 1 instead?

Alternatively, perhaps we should rename the overall function to `is_d_separator_minimal`, so that way it implies that one already has a d-separator. 

In summary:
1. I could rename the function
2. I could remove the `check_d_sep` kwarg
3. Or do something else.

WDYT?

Reply:
FWIW my [original opinion](https://github.com/networkx/networkx/issues/6216#issuecomment-1335938007) hasn't changed:
 1. I'm -1 on a keyword argument. Given the function name, most users will assume that the function checks that a d-separator is both valid **and** minimal. Adding a keyword argument requires the users to study the API in greater detail (i.e. look at each parameter in detail and read the docstring) to find out this isn't the case, and thus is poor API design IMO.
 2. Though it might help the situation, I'm also -1 on adding a new function name. We can't get rid of `is_minimal_d_separator` without a deprecation period, and IMO it'd be too confusing to have **both** `is_minimal_d_separator` and `is_d_separator_minimal`, even if there is a deprecation warning indicating which should be used.

Given the above, I think the course of action that makes the most sense remains to add the `d_separated` check internally to to the `is_minimal_d_separator` function. Again, see my original comment for more detailed explanation of the rationale, but given the API already in place, this is the course of action that is minimally disruptive. The cost of course is that `is_minimal_d_separator` is unnecessarily expensive in the case where a user already knows the nodes are `d_separated`. To me, a clear & simple user interface is worth the performance hit.

Reply:
I think I'm spinning our wheels in this discussion. Sorry about that. And Thanks @adam2392 and @rossbar for making it more clear to me what's going on.

I'm now OK with the original proposal without the keyword and with always verifying that `z` is a d_separator.
Sorry @adam2392 but could you remove the keyword again and get it back to the API you originally had?
I do like the additional tests you added and doc_string changes other than for the keyword...
:}

Thanks!

Reply:
Done in https://github.com/networkx/networkx/pull/6427/commits/5bd573a6ead6c4482f812616790b68effb00ccfe!